src/ziggurat/messaging/rabbitmq/cluster/producer.clj

```diff
@@ -13,30 +13,39 @@
 
 (defn get-default-ha-policy [cluster-config replica-count]
   (let [ha-mode      (get cluster-config :ha-mode "exactly")
-        ha-params    (get cluster-config :ha-params  replica-count)
+        ha-params    (get cluster-config :ha-params replica-count)
         ha-sync-mode (get cluster-config :ha-sync-mode "automatic")]
     (if (= "all" ha-mode)
       {:ha-mode "all" :ha-sync-mode ha-sync-mode}
       {:ha-mode ha-mode :ha-sync-mode ha-sync-mode :ha-params ha-params})))
 
+(defn set-ha-policy-on-host [host-endpoint username password ha-policy-body exchange-name queue-name]
+  (try
+    (binding [lh/*endpoint* host-endpoint
+              lh/*username* username
+              lh/*password* password]
+      (log/info "applying HA policies to queue: " queue-name)
+      (log/info "applying HA policies to exchange: " exchange-name)
+      (lh/set-policy "/" (str queue-name "_ha_policy")
+                     {:apply-to   "all"
+                      :pattern    (str "^" queue-name "|" exchange-name "$")
+                      :definition ha-policy-body}))
+    true
```

Comment:
`la/set-policy` returns an HTTP response. was not aware of this earlier. Just checked the code.

we can use that in instead of returning a boolean. 

Probably, no change in loop-recur is required. the `status` at line 45 will simply be an HTTP response on successful call or a nil on an exception.